File: dsa/02-stack/stack.py
from sys import maxsize


def create_stack():
    stack = []
    return stack


def is_empty(stack):
    return len(stack) == 0


def push(stack, item):
    stack.append(item)


def pop(stack):
    if is_empty(stack):
        return str(-maxsize - 1)  # return minus infinite

    return stack.pop()


def peek(stack):
    if is_empty(stack):
        return str(-maxsize - 1)  # return minus infinite
    return stack[len(stack) - 1]


# Driver program to test above functions
stack = create_stack()
push(stack, str(10))
push(stack, str(20))
push(stack, str(30))
print(pop(stack) + " popped from stack")
print("stack after popping an element: " + str(stack))

File: dsa/02-stack/stack.cpp
#include <bits/stdc++.h>
#include <iostream>

using namespace std;
#define MAX_SIZE 5
class Stack
{
    int TOP;

public:
    int array[MAX_SIZE];
    Stack()
    {
        TOP = -1;
    }
    bool isFull()
    {
        if (TOP > (MAX_SIZE - 1))
            return true;
        return false;
    }
    bool isEmpty()
    {
        if (TOP < 0)
            return true;
        return false;
    }
    void push(int item)
    {
        if (!isFull())
        {
            // TOP++;
            array[++TOP] = item;
        }
    }
    int peek()
    {
        if (!isEmpty())
        {
            return array[TOP];
        }
        return 0;
    }
    int pop()
    {
        if (isEmpty())
        {
            cout << "stack is empty";
            return 0;
        }
        else
        {
            int x = array[TOP--];
            // TOP--;
            return x;
        }
    }
};

int main()
{
    class Stack s;
    s.push(10);
    s.push(20);
    s.push(30);
    cout << s.pop() << " Popped from stack\n";

    // print top element of stack after popping
    cout << "Top element is : " << s.peek() << endl;

    cout << "Stack example\n";
    while (!s.isEmpty())
    {
        cout << s.peek() << " ";
        // remove top element in stack
        s.pop();
    }
    return 0;
}